react-components/src/components/AuthBar/authBar.js

```diff
@@ -22,8 +22,13 @@ import { func } from 'prop-types';
 
 const AuthBar = props => {
     const { showSignIn, showMyAccount } = props;
+    const [{ currentUser, isSignedIn }, { getUserDetails }] = useUserContext();
 
-    const [{ currentUser, isSignedIn }] = useUserContext();
+    useEffect(() => {
+        if (isSignedIn && currentUser.email === '') {
+            getUserDetails();
+        }
+    }, [getUserDetails]);
```

Comment:
Is this `useEffect` section executed again if you logout and log in again? It looks like the `getUserDetails` function never changes, so this section is only executed once.

Reply:
Yes, it is executed again. The dependency function changes every time the `fetchUserDetails` function changes.